other_enrichment_methods/2_viz_pathway_ranking.py

```diff
@@ -0,0 +1,596 @@
+platform = "rnaseq"
+# -
+
+# Load params
+dataset_name = params["dataset_name"]
+project_id = params["project_id"]
+
+# Create dictionary of enrichment method: statistic
+method_stats_dict = {"GSEA": "padj", "GSVA": "ES", "CAMERA": "FDR", "ORA": "p.adjust"}
+
+# ## Get pathway summary data
+
+# Pathway summary files
+gsea_pathway_summary_filename = os.path.join(
+    base_dir, dataset_name, f"generic_pathway_summary_{project_id}.tsv"
+)
+gsva_pathway_summary_filename = os.path.join(
+    base_dir, dataset_name, f"generic_pathway_summary_{project_id}_GSVA.tsv"
+)
+camera_pathway_summary_filename = os.path.join(
+    base_dir, dataset_name, f"generic_pathway_summary_{project_id}_CAMERA.tsv"
+)
+ora_pathway_summary_filename = os.path.join(
+    base_dir, dataset_name, f"generic_pathway_summary_{project_id}_ORA.tsv"
+)
+
+# +
+# Pathway summary data
+gsea_pathway_summary = pd.read_csv(
+    gsea_pathway_summary_filename, sep="\t", index_col=0, header=0
+)
+gsva_pathway_summary = pd.read_csv(
+    gsva_pathway_summary_filename, sep="\t", index_col=0, header=0
+)
+camera_pathway_summary = pd.read_csv(
+    camera_pathway_summary_filename, sep="\t", index_col=0, header=0
+)
+
+if platform == "rnaseq":
+    ora_pathway_summary = pd.read_csv(
+        ora_pathway_summary_filename, sep="\t", index_col=0, header=0
+    )
+# -
+
+# ## Format data for plotting
+
+print(gsea_pathway_summary.shape)
+gsea_pathway_summary.head()
+
+print(gsva_pathway_summary.shape)
+gsva_pathway_summary.head()
+
+print(camera_pathway_summary.shape)
+camera_pathway_summary.head()
+
+if platform == "rnaseq":
+    print(ora_pathway_summary.shape)
+    ora_pathway_summary.head()
+
+
+# ## Pathway ranking per enrichment method
+
+# define plotting function
+def plot_significance_vs_ranking(
+    summary_df, method_name, stats_colname, x_label, output_figure_filename
+):
+    # Format input dataframe
+
+    if stats_colname == "ES":
+        plot_df = pd.DataFrame(
+            data={
+                "Test statistic": summary_df[
+                    f"Median {stats_colname} (simulated)"
+                ].values,
+                "Percentile rank": summary_df["Percentile (simulated)"].values,
+            },
+            index=summary_df.index,
+        )
+    else:
+        plot_df = pd.DataFrame(
+            data={
+                "Test statistic": -np.log10(
+                    summary_df[f"Median {stats_colname} (simulated)"].values
+                ),
+                "Percentile rank": summary_df["Percentile (simulated)"].values,
+            },
+            index=summary_df.index,
+        )
+
+    fig = pn.ggplot(plot_df, pn.aes(x="Test statistic", y="Percentile rank"))
+    fig += pn.geom_point()
+    fig += pn.geom_point(
+        plot_df[plot_df["Percentile rank"] > 90.0],
+        pn.aes(x="Test statistic", y="Percentile rank"),
+        color="red",
+    )
+
+    # Only used for array data
+    # Manually stagger labels since they overlap here
+    """
```

Comment:
Are you using this function for array data at all? If so, you might want to pass in an argument for array or RNA-seq, then run the relevant code block.

Reply:
This section is commented out since we only decided to show the results for the RNA-seq data since the array didn't produce a curve for ORA because there were some simulated experiments that didn't find any DEGs. I'll go ahead and add this params for future use